### PR TITLE
New Transaction Builder/sender

### PR DESCRIPTION
**Pull request by deadeagle63**
Feature/new txns

## Types of changes

- [x] Technical Debt (non-breaking change which removes unused code/assets)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [ ] The related ClickUp task has been linked to this PR
- [ ] The person creating the pull request is listed in "Assignees"
- [ ] Change requires updated documentation

### CODE REVIEW

File: src/hooks/useTransaction.tsx [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+type useTransactionReturn = {
+  createTransactionBuilder: (txn?: TXN) => TransactionBuilder
+  createTransaction: (txn?: TXN) => Transaction
+  sendTransaction: (
+    txn: Transaction,
+    {
+      connection,
+      options,
+      confirmationWaitType
+    }?: {
```
Comment (AlexOfc) [inline src/hooks/useTransaction.tsx]: You have same type on line 47, can u move this copy paste to separate type variable?

File: src/hooks/useTransaction.tsx [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+function useTransaction(): useTransactionReturn {
+  const { priorityFeeValue } = useConnectionConfig()
+  const { sendTransaction: sendTransactionOriginal } = useWallet()
+  const { connection: originalConnection } = useConnectionConfig()
+
+  const createTransactionBuilder = useCallback(
+    (txn?: TXN) => new TransactionBuilder(txn).setPriorityFee(priorityFeeValue),
+    [priorityFeeValue]
+  )
+  const createTransaction = useCallback(
```
Comment (AlexOfc) [inline src/hooks/useTransaction.tsx:37]: do u need this method? it is not used anywhere

**Review by AlexOfc — CHANGES_REQUESTED** [review]

File: src/hooks/useTransaction.tsx (hunk 1 above)
Comment (deadeagle63) [reply src/hooks/useTransaction.tsx]: Segmented out.

File: src/hooks/useTransaction.tsx (hunk 2 above)
Comment (deadeagle63) [reply src/hooks/useTransaction.tsx:37]: Added as an additional optional builder if someone doesn't want the builder but just the txn for example

**Review by walt-1 — APPROVED** [review]

**Pull request merged**